adding support for OpenBSD distribution

serge-p opened the pull request:
## adding support for OpenBSD distribution into salt-bootstrap.sh
- added install functions for OpenBSD distribution into salt-bootstrap.sh
- added my contact into into AUTHORS.rst

serge-p committed:
adding support for OpenBSD

jfindlay commented:
Excellent, thanks, @serge-p.

serge-p commented:
Thanks, @jfindlay any chance to merge changes ?

jfindlay commented:
@serge-p, yes, definitely.  I am going to be spending time on bootstrap today and hope to get everything merged.

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -586,6 +586,7 @@ __fetch_url() {
         wget $_WGET_ARGS -q -O "$1" "$2" >/dev/null 2>&1 ||
             fetch $_FETCH_ARGS -q -o "$1" "$2" >/dev/null 2>&1 ||
                 fetch -q -o "$1" "$2" >/dev/null 2>&1           # Pre FreeBSD 10
```
@serge-p, should there be an `||` at the end of this line?

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -4325,6 +4353,179 @@ install_freebsd_restart_daemons() {
+
+install_openbsd_deps() {
+  ping -c 2 -q mirrors.nycbug.org || return 1
+  echo "installpath = http://mirrors.nycbug.org/pub/OpenBSD/${OS_VERSION}/packages/${CPU_ARCH_L}/" >/etc/pkg.conf || return 1
```
@serge-p, is there a reason this is dependent on a specific mirror?

serge-p replied:
@jfindlay, sure, my reason was  nycbug is closest to me official US mirror  
can use any official openbsd http mirror from the [list](http://www.openbsd.org/ftp.html#http) 
for example [ftp5.openbsd.org](http://ftp5.usa.openbsd.org/pub/OpenBSD/)
or can add more mirrors into the list

jfindlay replied:
Is it possible to reference all mirrors in a generic manner?  If not, I guess this will be fine.

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -4325,6 +4353,179 @@ install_freebsd_restart_daemons() {
+      # Let's set the configuration directory to /tmp
+      _TEMP_CONFIG_DIR="/tmp"
+      CONFIG_SALT_FUNC="config_salt"
+      for fname in minion master; do
```
@serge-p, `for fname in minion master syndic; do`?

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -4325,6 +4353,179 @@ install_freebsd_restart_daemons() {
+          # Let's download, since they were not provided, the default configuration files
+          if [ ! -f "$_SALT_ETC_DIR/$fname" ] && [ ! -f "$_TEMP_CONFIG_DIR/$fname" ]; then
+              ftp -o "$_TEMP_CONFIG_DIR/$fname" \
+                  "https://raw.githubusercontent.com/saltstack/salt/develop/conf/$fname" || return 1
```
@serge-p, this should probably be using `$STABLE_REV`: `https://raw.githubusercontent.com/saltstack/salt/${STABLE_REV}/conf/$fname`.

serge-p replied:
@jfindlay, just checked, and this link is the same for SmartOS and SuSE Install Functions

jfindlay commented on bootstrap-salt.sh:
```diff
@@ -4325,6 +4353,179 @@ install_freebsd_restart_daemons() {
+        [ $fname = "master" ] && [ "$_INSTALL_MASTER" -eq $BS_FALSE ] && continue
+        [ $fname = "api" ] || [ "$(which salt-${fname} 2>/dev/null)" = "" ] && continue
+        #[ $fname = "api" ] && ([ "$_INSTALL_MASTER" -eq $BS_FALSE ] || [ "$(which salt-${fname} 2>/dev/null)" = "" ]) && continue
+        #[ $fname = "syndic" ] && [ "$_INSTALL_SYNDIC" -eq $BS_FALSE ] && continue
```
@serge-p, can we remove these commented lines?  Are they not needed?

serge-p replied:
@jfindlay, these lines are important as a reference, pointing on a corresponding change in openbsd source tree. Would it be a big deal to keep them in place ?

vutny replied:
@serge-p By the way, we have `__check_command_exists` function in the script. Better to use it instead of `which`, like this:

```
[ $fname = "api" ] || __check_command_exists salt-${fname} && continue
```

serge-p committed:
Update bootstrap-salt.sh

corrected for __fetch_url() and install_openbsd_deps()

serge-p committed:
Update bootstrap-salt.sh

added choose_openbsd_mirror() function, 
which is comparing ping time from all officially listed openbsd mirrors 
and picks closest mirror.

serge-p committed:
Update bootstrap-salt.sh

removed commented unneeded lines from install_openbsd_post() function

jfindlay commented:
@vutny, I will take care of the which conversions in a separate pull request.  There are a few stray uses beyond OpenBSD, it seems.

jfindlay commented:
@serge-p, thanks for your great work and patience.
